src/Common/AsynchronousMetrics.h

```diff
@@ -100,14 +101,18 @@ class AsynchronousMetrics
     std::condition_variable wait_cond;
     bool quit TSA_GUARDED_BY(thread_mutex) = false;
 
+    /// Protects all raw data and serializes multiple updates.
     mutable std::mutex data_mutex;
 
     /// Some values are incremental and we have to calculate the difference.
     /// On first run we will only collect the values to subtract later.
     bool first_run TSA_GUARDED_BY(data_mutex) = true;
     TimePoint previous_update_time TSA_GUARDED_BY(data_mutex);
 
-    AsynchronousMetricValues values TSA_GUARDED_BY(data_mutex);
+    /// Protects saved values.
+    mutable SharedMutex values_mutex;
+    /// Values store the result of the last update prepared for reading.
+    AsynchronousMetricValues values TSA_GUARDED_BY(values_mutex);
```

Comment:
```suggestion
#ifdef OS_LINUX
    AsynchronousMetricValues values TSA_GUARDED_BY(values_mutex);
#else
    /// When SharedMutex == std::shared_mutex it may not be annotated with the 'capability'.
    AsynchronousMetricValues values;
#endif
```